internal/validator/storage.go

```diff
@@ -91,5 +93,13 @@ func readFile(reader io.Reader) (result []byte, err error) {
 		}
 	}
 
+	if compression == "xz" {
+		if xzReader, err := xz.NewReader(reader, 0); err != nil {
```

Comment:
if you assign to reader directly you can leave out the else branch entirely. For gzip it's needed for the Close() call but here it's not